x-pack/plugins/case/server/services/index.ts

```diff
@@ -0,0 +1,193 @@
+  commentId: string;
+  updatedAttributes: UpdatedCommentType;
+}
+
+interface GetUserArgs {
+  request: KibanaRequest;
+  response: KibanaResponseFactory;
+}
+
+interface CaseServiceDeps {
+  authentication: SecurityPluginSetup['authc'] | null;
+}
+export interface CaseServiceSetup {
+  deleteCase(args: GetCaseArgs): Promise<{}>;
+  deleteComment(args: GetCommentArgs): Promise<{}>;
+  getAllCases(args: ClientArgs): Promise<SavedObjectsFindResponse>;
+  getAllCaseComments(args: GetCaseArgs): Promise<SavedObjectsFindResponse>;
+  getCase(args: GetCaseArgs): Promise<SavedObject>;
+  getComment(args: GetCommentArgs): Promise<SavedObject>;
+  getUser(args: GetUserArgs): Promise<any>;
+  postNewCase(args: PostCaseArgs): Promise<any>;
+  postNewComment(args: PostCommentArgs): Promise<any>;
+  updateCase(args: UpdateCaseArgs): Promise<any>;
+  updateComment(args: UpdateCommentArgs): Promise<any>;
+}
+
+export class CaseService {
+  constructor(private readonly log: Logger) {}
+  public setup = async ({ authentication }: CaseServiceDeps): Promise<CaseServiceSetup> => ({
+    deleteCase: async ({ client, caseId }: GetCaseArgs) => {
+      try {
+        this.log.debug(`Attempting to GET case ${caseId}`);
+        return await client.delete(CASE_SAVED_OBJECT, caseId);
+      } catch (error) {
+        this.log.debug(`Error on GET case ${caseId}: ${error}`);
+        throw error;
+      }
+    },
+    deleteComment: async ({ client, commentId }: GetCommentArgs) => {
+      try {
+        this.log.debug(`Attempting to GET comment ${commentId}`);
+        return await client.delete(CASE_COMMENT_SAVED_OBJECT, commentId);
+      } catch (error) {
+        this.log.debug(`Error on GET comment ${commentId}: ${error}`);
+        throw error;
+      }
+    },
+    getCase: async ({ client, caseId }: GetCaseArgs) => {
+      try {
+        this.log.debug(`Attempting to GET case ${caseId}`);
+        return await client.get(CASE_SAVED_OBJECT, caseId);
+      } catch (error) {
+        this.log.debug(`Error on GET case ${caseId}: ${error}`);
+        throw error;
+      }
+    },
+    getComment: async ({ client, commentId }: GetCommentArgs) => {
+      try {
+        this.log.debug(`Attempting to GET comment ${commentId}`);
+        return await client.get(CASE_COMMENT_SAVED_OBJECT, commentId);
+      } catch (error) {
+        this.log.debug(`Error on GET comment ${commentId}: ${error}`);
+        throw error;
+      }
+    },
+    getAllCases: async ({ client }: ClientArgs) => {
+      try {
+        this.log.debug(`Attempting to GET all cases`);
+        return await client.find({ type: CASE_SAVED_OBJECT });
+      } catch (error) {
+        this.log.debug(`Error on GET cases: ${error}`);
+        throw error;
+      }
+    },
+    getAllCaseComments: async ({ client, caseId }: GetCaseArgs) => {
+      try {
+        this.log.debug(`Attempting to GET all comments for case ${caseId}`);
+        return await client.find({
```

Comment:
@XavierM this is how I'm connecting the case to the comment. I see the `references` field though. Do you think I should be using that in conjunction with the `has_reference` argument on a `find`?

Reply:
yes, I will use it, it is part of the API and if you look at [the code](https://github.com/elastic/kibana/blob/master/src/core/server/saved_objects/service/lib/search_dsl/query_params.ts#L117), it seems to be built for that. So now the question is why I did not use it for timeline and notes simply because I went to fast and did not see it.